### PR TITLE
Complete create_stop functionality

### PR DESCRIPTION
**Pull request by maximsolodukhin**
- Completed logic of bus line (bus_id) selection when creating a new dynamic stop
- Adapted columns in supabase to our functionality
- Modified response bodies of "create_stop"
- Modified supabase calls accordingly
- supabase.rpc('create_stop') creates new "stop" and "bus_stop_mapping"

### CODE REVIEW

File: routers/stops.py [hunk 1]
```diff
@@ -5,43 +5,48 @@
 
 router = APIRouter(prefix="/stops", tags=["stops"])
 
-#TODO please check the logic here
-#TODO while creating a stop as an operator, I get a bad request error but stop is still created
 # Define the endpoint for creating a stop
 @router.post("/stop")
```
Comment (nikita-kostin) [inline routers/stops.py]: Missed this before, the URL suffix should be just "/"

File: routers/stops.py [hunk 2]
```diff
@@ -1,66 +1,86 @@
...
+def create_stop(stop : Stop, current_user: User = Depends(get_current_user)):
+    # Check if the current user is a driver
+    if current_user.entity == UserEntity.driver:
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN,
+            detail="Drivers can't create stops",
+        )
+    else:
+        if stop.entity == StopEntity.static:
+            if current_user.entity != UserEntity.admin:
```
Comment (nikita-kostin) [inline routers/stops.py]: Who admin

Comment (nikita-kostin) [reply routers/stops.py]: I understand the concern that we don't really have a user type to create the bus stops, but I'd rather tie this functionality to parcel manager

Comment (nikita-kostin) [reply routers/stops.py]: In any case the name admin is probably too ambiguous

File: routers/stops.py [hunk 3]
```diff
@@ -1,66 +1,86 @@
...
-    # Check if the current user is a passenger or a parcel operator
-    if current_user.entity == UserEntity.passenger or current_user.entity == UserEntity.parcel_operator:
-        if current_user.entity == UserEntity.passenger:
-            # Check if the passenger already requested a stop
-            # Query the stop table with user id
-            response = supabase.table("stops").select("*").eq("user_id", current_user.id).execute()
-            # Check if response has data
-            if response.data:
-                # Raise an exception if the stop is not possible
+def create_stop(stop : Stop, current_user: User = Depends(get_current_user)):
```
Comment (nikita-kostin) [inline routers/stops.py:10]: I'd rather pass separate parameters than the whole model

File: routers/stops.py [hunk 4]
```diff
@@ -1,66 +1,86 @@
...
+                    stop.bus_id = nearest_bus_id
+                    break
+            if stop.bus_id is None:
+                # Raise an exception if the stop is not possible
+                # TODO: (optional) change to another status code, e.g., 2xx
+                raise HTTPException(
+                    status_code=status.HTTP_400_BAD_REQUEST,
+                    detail="Stop creation is not possible: no bus lines nearby",
+                )
         # Insert the stop data into the stop table
```
Comment (nikita-kostin) [inline routers/stops.py]: Looks rather complicated, we should ideally derive this functionality into functions

File: routers/stops.py [hunk 5]
```diff
@@ -69,15 +89,14 @@ def list_stops(current_user: User = Depends(get_current_user)):
         # Return an empty list if no stops are found
         return []
 
-#TODO does not depend on the user
+#TODO doesn't need to be an endpoint, only used inside create_stop
 @router.get("/stops_sorted")
-def get_stops_sorted(current_user: User = Depends(get_current_user), lat: float = 0, long: float = 0):
```
Comment (nikita-kostin) [inline routers/stops.py:74]: I'm not 100% sure, but it's possible that by removing the user parameter completely we'll allow unauthorised access to the endpoint. Do we want this?

File: routers/stops.py [hunk 6]
```diff
@@ -69,15 +89,14 @@ def list_stops(current_user: User = Depends(get_current_user)):
...
-    response = supabase.rpc('nearby_stops', {"lat": lat, "long": long}).execute()
+def get_stops_sorted(lat: float = 0, long: float = 0):
+    response = supabase.rpc('nearby_stops', {"lat_position": lat, "long_position": long}).execute()
     return {"stops": response.data}
 
 
-#TODO does not depend on the user
 @router.get("/stops_in_range")
-def get_stops_in_range(current_user: User = Depends(get_current_user), min_lat: float = 0, min_long: float = 0, max_lat: float = 0, max_long: float = 0):
```
Comment (nikita-kostin) [inline routers/stops.py:81]: Same as above

File: routers/stops.py (hunk 3 above)
Comment (maximsolodukhin) [reply routers/stops.py:10]: L + ratio

**Review by nikita-kostin — APPROVED** [review]
🚀

**Pull request merged**